ui/models/browsertimeAlertsExtraData.js

```diff
@@ -93,32 +75,34 @@ export default class BrowsertimeAlertsExtraData {
     return RepositoryModel.getRepo(this.alertSummary.repository, repos);
   }
 
-  async enrichWithProfileLinks(alertSummary, repo, jobList) {
+  async enrichWithProfileLinks(alertSummary, repo) {
     let alertLinks = alertSummary.alerts.map((alert) => {
-      const job = jobList.data.find(
-        (j) =>
-          j.searchStr.includes(alert.series_signature.suite) &&
-          j.searchStr.includes(alert.series_signature.machine_platform) &&
-          j.resultStatus === 'success',
-      );
-
-      if (job) {
+      if (alert.taskcluster_metadata && alert.prev_taskcluster_metadata) {
         const { suite } = alert.series_signature;
 
-        const url = getArtifactsUrl({
-          taskId: job.task_id,
-          run: job.retry_id,
+        const currentUrl = getArtifactsUrl({
+          taskId: alert.taskcluster_metadata.task_id,
+          run: alert.taskcluster_metadata.retry_id,
+          rootUrl: repo.tc_root_url,
+          artifactPath: `public/test_info/profile_${suite}.zip`,
+        });
+        const prevUrl = getArtifactsUrl({
+          taskId: alert.prev_taskcluster_metadata.task_id,
+          run: alert.prev_taskcluster_metadata.retry_id,
           rootUrl: repo.tc_root_url,
           artifactPath: `public/test_info/profile_${suite}.zip`,
         });
 
         // We check the artifacts with `method: 'HEAD'` which means that it
         // doesn't download the whole file and only gets the headers. We can see
         // if artifact is available or not using the headers.
-        const promise = fetch(url, {
+        const currentPromise = fetch(currentUrl, {
+          method: 'HEAD',
+        });
+        const prevPromise = fetch(prevUrl, {
```

Comment:
There is a bit of code duplication here. We could probably extract that part to its own local function. But it's probably fine to leave it like this too since it's only for 2 urls.